### PR TITLE
Implement password recovery functionality on backend

### PR DESCRIPTION
**Pull request by munteanumarco**
I tested password recovery functionality manually with swagger.
Added a mail service (this can be further reused for account confimation functionality when a new user is registered)
I also configured Serilog as our logging solution.

### CODE REVIEW

File: EventPlannerAPI/BusinessLayer/Services/UserLogicService.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                var user = await _userService.FindByEmailAsync(forgotPasswordDto.Email);
+
+                if (user == null)
+                {
+                    _logger.Error($"Error sending reset link: User with email {forgotPasswordDto.Email} does not exist");
+                    return string.Empty;
+                }
+
+                var token = await _userService.GeneratePasswordResetTokenAsync(user);
+                var resetLink = "https://eventplanner.com/reset-password?token=" + token;
```
Comment (alexandratodea) [inline EventPlannerAPI/BusinessLayer/Services/UserLogicService.cs]: will this work on localhost?

Comment (munteanumarco) [reply EventPlannerAPI/BusinessLayer/Services/UserLogicService.cs]: yeah, should be the url of the react app, for now I put this just as a placeholder and I will change it when I will integrate with the frontend

Comment (anepher) [reply EventPlannerAPI/BusinessLayer/Services/UserLogicService.cs]: This url should not be into the code but rather config file

Comment (alexandratodea) [reply EventPlannerAPI/BusinessLayer/Services/UserLogicService.cs]: you could add the react app URL intro an environment variable and use it here

**Review by alexandratodea — APPROVED** [review]

File: EventPlannerAPI/BusinessLayer/Services/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                var user = await _userService.FindByEmailAsync(forgotPasswordDto.Email);
+
+                if (user == null)
+                {
+                    _logger.Error($"Error sending reset link: User with email {forgotPasswordDto.Email} does not exist");
+                    return string.Empty;
+                }
+
+                var token = await _userService.GeneratePasswordResetTokenAsync(user);
+                var baseUrl = _configuration["FrontendBaseUrl"];
```
Comment (gheorghe-sava) [inline EventPlannerAPI/BusinessLayer/Services/UserService.cs]: We need to create a separate static class with configuration strings

Comment (gheorghe-sava) [inline EventPlannerAPI/BusinessLayer/Services/UserService.cs]:
```suggestion
                var baseUrl = _configuration[SolutionConfigurationConstants.FrontendBaseUrl];
```

**Review by gheorghe-sava — APPROVED** [review]

**Pull request merged**